pkg/reconciler/dependencybuild/dependencybuild.go

```diff
@@ -429,8 +429,33 @@ func (r *ReconcileDependencyBuild) handleStateAnalyzeBuild(ctx context.Context,
 		}
 
 		db.Status.PotentialBuildRecipes = buildRecipes
-		db.Status.State = v1alpha1.DependencyBuildStateSubmitBuild
+
+		jbsConfig := &v1alpha1.JBSConfig{}
+		err := r.client.Get(ctx, types.NamespacedName{Namespace: db.Namespace, Name: v1alpha1.JBSConfigName}, jbsConfig)
+		if err != nil && !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		// As the tests utilise prependTag for uniqueness (See also
+		// build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/deploy/Gav.java) so check for that
+		// here to avoid reusing images when we want differentiation.
+		prependTag := len(jbsConfig.Spec.PrependTag) > 0
```

Comment:
So the problem with this is that then we won't be able to test that this is working. 

If this is set images should be tagged with prependTag + imageId, and the lookup command should be passed it as a parameter to look for prependTag + imageId.